synapse/rest/media/v1/storage_provider.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2023 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This exists purely for backwards compatibility with media providers.
+from synapse.media.storage_provider import StorageProvider  # noqa: F401
```

Comment:
Is this necessary in the light of https://github.com/matrix-org/synapse/pull/15146/files#diff-93cc7d2bb16bade8a31d19c3f919055b5c93490fb5ab935c5486aeb87178f335L181-R187 ? I suppose it is if some modules have been importing from here.

Reply:
Yeah, pretty much. I'm not really sure what the "real" fix is there (and I think the S3 provider is the only 3rd party one, but I'm not really sure).